CMakeLists.txt

```diff
@@ -366,28 +372,82 @@ else()
 
   option(NO_AES "Explicitly disable AES support" ${NO_AES})
 
-  if(NOT NO_AES AND NOT (ARM6 OR ARM7))
+  if(NOT NO_AES AND NOT (ARM6 OR ARM7 OR ARM8))
     message(STATUS "AES support enabled")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -maes")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -maes")
+  elseif(ARM8)
+    message(STATUS "AES support enabled for ARMv8")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=armv8-a+crypto")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=armv8-a+crypto")
   elseif(ARM7 OR ARM6)
-    message(STATUS "AES support disabled (not available on ARM)")
+    message(STATUS "AES support disabled (not available on ARMv6 or ARMv7)")
   else()
     message(STATUS "AES support disabled")
   endif()
 
   if(ARM6)
-    message(STATUS "Setting ARM6 C and C++ flags")
+    message(STATUS "Setting ARMv6 C and C++ flags")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfpu=vfp -mfloat-abi=hard")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfpu=vfp -mfloat-abi=hard")
   endif()
 
   if(ARM7)
-    message(STATUS "Setting ARM7 C and C++ flags")
+    message(STATUS "Setting ARMv7 C and C++ flags")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfloat-abi=hard")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfloat-abi=hard")
+  endif()
+
+  if(ARM8)
+  include(TestCXXAcceptsFlag) #for Raspberry Pi 3
+  CHECK_CXX_ACCEPTS_FLAG(-mfpu=neon-fp-armv8 CXX_ACCEPTS_NEON_FP_ARMV8)
+  if(CXX_ACCEPTS_NEON_FP_ARMV8)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfpu=neon-fp-armv8 -funsafe-math-optimizations")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfpu=neon-fp-armv8 -funsafe-math-optimizations")
+  endif()
+  CHECK_CXX_ACCEPTS_FLAG(-mfloat-abi=hard CXX_ACCEPTS_MFLOAT_HARD)
+  if (CXX_ACCEPTS_MFLOAT_HARD)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfloat-abi=hard")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfloat-abi=hard")
   endif()
 
+  message(STATUS "Attempting to tune for specific ARMv8 Architecture")
+  execute_process(COMMAND taskset -c 1 grep "^CPU part" /proc/cpuinfo OUTPUT_VARIABLE PROC_PART)
+  if (PROC_PART MATCHES "0xd04") #Cortex-A35
+    message(STATUS "Tuning for Cortex-A35")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mtune=cortex-a35")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mtune=cortex-a35")
+  elseif (PROC_PART MATCHES "0xd07") #Cortex-A57
+    message(STATUS "Tuning for Cortex-A57")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mtune=cortex-a57")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mtune=cortex-a57")
+  elseif (PROC_PART MATCHES "0xd08") #Cortex-A72
+    message(STATUS "Tuning for Cortex-A72")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mtune=cortex-a72")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mtune=cortex-a72")
+  elseif (PROC_PART MATCHES "0xd03") #Cortex-A53
+    message(STATUS "Tuning for Cortex-A53")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mtune=cortex-a53")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mtune=cortex-a53")
```

Comment:
Please remove all -mtune setting code (see my comment in the comment section of this PR).